ui/pages/confirmations/components/simulation-details/fiat-display.tsx

```diff
@@ -55,6 +61,11 @@ export const IndividualFiatDisplay: React.FC<{ fiatAmount: FiatAmount }> = ({
 export const TotalFiatDisplay: React.FC<{
   fiatAmounts: FiatAmount[];
 }> = ({ fiatAmounts }) => {
+  const hideFiatForTestnet = useHideFiatForTestnet();
+  if (hideFiatForTestnet) {
```

Comment:
Do we need to move this below all the other hooks to ensure they are always run per render?

Same in the above component also.

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/25167/commits/c6c0ddbe73f24d1bfdf7fe30af01a6cd025ef837